### PR TITLE
Add point-of-view entity and component to the dataframe view's UI

### PR DESCRIPTION
**Pull request by abey79**
### What

- Part of: https://github.com/rerun-io/rerun/issues/7279

This PR updates the dataframe query override UI as per design in #7279, in particular adding PoV entity and component.
- updated UI layout
- time boundaries default to `+∞`/`–∞` button which, when clicked, turn into editable time drag value
- reset buttons to go back to the `∞` state
- auto-selection of PoV component based on PoV entity (picks a required component for one of the entity archetypes)

**Note**:
- This is a pure UI PR. The PoV entity/component are not yet used at all for the dataframe's content (that will be addressed in a follow-up PR currently blocked on #7284).
- Ignore the ugly "Time range table order" part, this will be cleaned up later (#7070)

https://github.com/user-attachments/assets/32151a1f-b0ca-4e99-99df-ea730451d4dc

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7331?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7331?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7331)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/store/re_types/definitions/rerun/blueprint/datatypes/time_range_query.fbs [hunk 1]
```diff
@@ -10,9 +10,15 @@ table TimeRangeQuery (
   /// Name of the timeline this applies to.
   timeline: rerun.datatypes.Utf8 (order: 100);
 
+  /// Point-of-view entity.
+  pov_entity: rerun.datatypes.EntityPath (order: 200);
+
+  /// Point-of-view component.
+  pov_component: rerun.datatypes.Utf8 (order: 300);
```
Comment (emilk) [inline crates/store/re_types/definitions/rerun/blueprint/datatypes/time_range_query.fbs:17]: I feel these deserve a bit more explanation

File: crates/viewer/re_space_view_dataframe/src/query_kind.rs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+        pov_component: ComponentName,
+        from: TimeInt,
+        to: TimeInt,
+    },
+    //TODO(#7067): add selected components
+}
+
+impl QueryKind {
+    /// Show the UI for the query kind selector.
+    pub(crate) fn ui(
```
Comment (emilk) [inline crates/viewer/re_space_view_dataframe/src/query_kind.rs:25]: should document the return value

**Review by emilk — APPROVED** [review]

**Comment by emilk** [comment]
The "Time range table order" has more indentation than the rest
image

**Comment by abey79** [comment]
> The "Time range table order" has more indentation than the rest image

I'm gonna let that one slide. I don't have control on that bit of UI (auto-derived from blueprint archetype), and it's going away anyways:
- #7070

**Pull request merged**